libsolidity/experimental/analysis/DebugWarner.cpp

```diff
@@ -39,19 +39,12 @@ bool DebugWarner::analyze(ASTNode const& _astRoot)
 
 bool DebugWarner::visitNode(ASTNode const& _node)
 {
-	auto const& typeInferenceAnnotation = m_analysis.annotation<TypeInference>(_node);
-	if (typeInferenceAnnotation.type)
-	{
-		Type type = *typeInferenceAnnotation.type;
-		Sort sort = m_analysis.typeSystem().env().sort(type);
-		std::string sortString;
-		if (sort.classes.size() != 1 || *sort.classes.begin() != m_analysis.typeSystem().primitiveClass(PrimitiveClass::Type))
-			sortString = ":" + TypeSystemHelpers{m_analysis.typeSystem()}.sortToString(m_analysis.typeSystem().env().sort(type));
+	std::optional<Type> const& inferredType = m_analysis.annotation<TypeInference>(_node).type;
+	if (inferredType.has_value())
```

Comment:
As for random style remarks: I find ``.has_value()`` on ``optional``s horrible myself personally :-D... but it doesn't matter at all, ultimately everybody can read both, so I don't bother pushing my own preferred style through.

Reply:
For me it's `if (inferredType)` that's horrible. It does not read like a condition at all. It forces me stop and think in which way this can be interpreted as a boolean, while on seeing `inferredType.has_value()` I immediately know it's an `optional` and can move on smoothly. I'm really not sure why people don't find this jarring.

Reply:
That's my point, though - it's purely subjective and we spend way too much time and effort in general in flipping back and forth around purely subjective taste that has no actual bearing on overall readability.

Reply:
As in: I find this significantly *less* readable with ``.has_value()`` than without - and I don't buy that it'd be worse for you to read it without that than it is for  me to read it with that - we should just generally have a big more flexibility around stuff like this.